app/main.py

```diff
@@ -33,23 +33,23 @@
     level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
 )
 
-
 class MainApp:
     def __init__(self):
         load_dotenv(override=True)
+        if os.getenv("isDevelopmentMode") == "enabled":
+            if os.getenv("ENV") != "development":
+                logging.error("Development mode enabled but ENV is not set to development. Exiting.")
+                exit(1)
         self.setup_sentry()
         self.app = FastAPI()
         self.setup_cors()
         self.initialize_database()
         self.check_and_set_env_vars()
-        if os.getenv("isDevelopmentMode") == "enabled":
-            self.setup_data()
-        else:
-            FirebaseSetup.firebase_init()
+        self.setup_data()
         self.include_routers()
 
     def setup_sentry(self):
-        if os.getenv("ENV") == "production":
```

Comment:
same problem here, removing this enables sentry in envs like stage as well. Should be prod only